### DISCUSSION TITLE
how to prevent run function inside script?

### DISCUSSION THREAD

**Discussion by nekooee** [uBlockOrigin/uAssets#16964] [Q&A] [answered]
Some sites try to find out that adblocker is activated on their site with different JavaScript methods.
For example, the "soft98.ir" tries to do this through the following file:
`https://soft98.ir/templates/22-09-20-18/js/jquery.min.packed.js`

I found out which of the functions inside this javascript file performs checks and detection of the adblocker. Now I want to prevent the execution of that method in this file. At the same time, I don't want to block the whole file.
How can this be done?

**Comment by MasterKia** [comment, marked as answer]
Hi, you need to know Javascript and write a code that defuses those anti-blocker codes.
For more information, see this issue:
https://github.com/MasterKia/PersianBlocker/issues/40

درود بر شما.
باید جاوااسکریپت بلد باشید و کدی بنویسید که اون کد ها رو خنثی کنه.
برای اطلاعات بیشتر این issue رو بخونید:
https://github.com/MasterKia/PersianBlocker/issues/40

**Reply by nekooee** [reply]
> درود بر شما. باید جاوااسکریپت بلد باشید و کدی بنویسید که اون کد ها رو خنثی کنه. برای اطلاعات بیشتر این issue رو بخونید: [MasterKia/PersianBlocker#40](https://github.com/MasterKia/PersianBlocker/issues/40)

سلام. تشکر از لطف شما. من جاوا اسکریپت بلد هستم منتها برای اجرای جاوا اسکریپت خودم میدونم که با افزونه های مختلفی میشه کد خودم رو پیوست سایت کنم. ولی آیا با خود  uBlockOrigin هم میشه اینکار رو کرد؟

**Reply by u-RraaLL** [reply]
@MasterKia Should you really reply in Persian to an English discussion on uAssets?

**Reply by nekooee** [reply]
> @MasterKia Should you really reply in Persian to an English discussion on uAssets?

sorry, He told me I had to do it with javascript. And I asked if there is a way to run my javascript code without needing another plugin and only with the uBlockOrigin?

**Reply by MasterKia** [reply]
@u-RraaLL 

What about:
https://github.com/uBlockOrigin/uAssets/issues/16882#issuecomment-1439852174

Could you add something useful instead of spam?

**Reply by MasterKia** [reply]
@nekooee Please send a message in this Telegram group:
https://t.me/PersianBlocker

**Reply by u-RraaLL** [reply]
> What about: [#16882 (comment)](https://github.com/uBlockOrigin/uAssets/issues/16882#issuecomment-1439852174)

That is a report (an issue) about a specific site. It was submitted in Japanese so a reply in Japanese makes perfect sense.

This here is general question (a discussion) with an example. It was asked in English. They are not the same.

> Could you add something useful

Herein lies my point. I've no idea what you said, so no - I can't. And if anybody else wanted to learn the same thing OP did, they wouldn't be able to as you've chosen to reply in another language.

Anyway, mapx- already replied in English so that point is now moot.

> instead of spam?

I don't consider asking one question regarding the content of this **discussion** to be spam.

I think if every volunteer started replying in their native language when it's not necessary, the whole repo would turn chaotic, bringing us much closer to actual spam.

**Comment by mapx-** [comment]
You need a scriptlet to do such things (replacing part of the code).

However, did you decrypt that obfuscated code ?

**Reply by mapx-** [reply]
For example, you should read the original script (read it via `fetch/XMLHttpRequest`), replace / remove that function - via regex - and append the new script to `document.body`

(and block using uBO the original one script)

https://stackoverflow.com/questions/58412644/load-external-js-file-modify-something-in-it-then-execute-it-normally
https://levelup.gitconnected.com/how-to-load-external-javascript-files-from-the-browser-console-8eb97f7db778

**Reply by nekooee** [reply]
> 



> You need a scriptlet to do such things (replacing part of the code).
> 
> However, did you decrypt that obfuscated code ?

Yes, decoding JavaScript files is not a difficult task. Because they have to be rendered and run anyway. Just read them from memory after it is executed and then at the beginning of the file, where the functions are named with strange names, understand them and change the variable names to readable names. You don't need to search all the files for this. The error text shown to you after AdBlocker plugin detection is a good start to finding the function you need to stop from there.

Now I did something simpler. First, I blocked JavaScript sources by testing them, until finally, I found out that this is the file that detects "adblocker extension". Then I blocked that file completely in my adblocker.
But that site kept asking you to activate your JavaScript. then I noticed that the links are in a section at the bottom of the site and only the max-height style is set to zero.

**Reply by krystian3w** [reply]
Still you have to remember that he can outsmart you to a few days and you will debug the code again.
